### PR TITLE
Add topic cleanup support

### PR DESCRIPTION
**Pull request by vnktram**
# Description

This change introduces a housekeeping endpoint to remove unused topics from Kafka

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

# Is it a breaking change?
No
How do we intend to rollback or handle backward compatibility in case something breaks?
Housekeeping endpoint that does not affect regular flow
Can this change be tested effectively via canary?
No
## Screenshots:

Include a screenshot of the changes, if applicable.

# Checklist Exception

- [ ] My change doesn't require the below checklist to be updated.
Please add a description on why it is not applicable.

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added unit tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added integration tests for the new feature (if applicable)
- [x] I have manually tested my code to the best of my abilities.

### CODE REVIEW

File: service/web/adminserver.go [hunk 1]
```diff
@@ -254,6 +255,68 @@ func (s adminServer) MigrateSubscriptions(ctx context.Context, subscriptions *me
...
+		if err != nil {
+			logger.Ctx(ctx).Errorw("Failed to fetch topics for project from messagebroker", "project", p)
+			continue
+		}
+
+		for t := range allTopics {
+			t = denormalizeTopicName(t)
+			if _, ok := validTopics[t]; !ok {
+				tops.Names = append(tops.Names, t)
+				dtresp, err := admin.DeleteTopic(ctx, messagebroker.DeleteTopicRequest{Name: t})
```
Comment (sakshi-goyal-razorpay) [inline service/web/adminserver.go:307]: do we have a bulk endpoint to delete topics..?

Comment (vnktram) [reply service/web/adminserver.go:307]: Kafka pkg supports bulk delete but deleting multiple topics in one shot could cause a lot of rebalancing for the existing topics. Want to do this sequentially.

File: service/web/adminserver.go [hunk 2]
```diff
@@ -254,6 +255,68 @@ func (s adminServer) MigrateSubscriptions(ctx context.Context, subscriptions *me
...
+		}
+		validTopics := make(map[string]bool, 0)
+		for _, sub := range subs {
+
+			for _, v := range sub.GetDelayTopics() {
+				validTopics[v] = true
+			}
+			validTopics[sub.GetDeadLetterTopic()] = true
+			validTopics[sub.GetRetryTopic()] = true
+			// validTopics[sub.GetSubscriptionTopic()] = true ###Internal topics are not used and hence up for deletion
```
Comment (mehuled) [inline service/web/adminserver.go:280]: Why is the SubscriptionTopic up for deletion? Isn't this the primary topic of the subscription.

Comment (vnktram) [reply service/web/adminserver.go:280]: Subscription topics are internal topics that were created for a re-arch but the re-arch plan was dropped. Primary topics are not subscriptions topics.

**Review by sakshi-goyal-razorpay — APPROVED** [review]

**Pull request merged**